numpy/distutils/system_info.py

```diff
@@ -1911,31 +1911,36 @@ def calc_info(self):
             # Use the system BLAS from Accelerate or vecLib under OSX
             args = []
             link_args = []
-            if get_platform()[-4:] == 'i386' or 'intel' in get_platform() or \
+            arm = intel = 0
+            if 'arm64' in get_platform() or 'aarch64' in get_platform():
+                arm = 1
+            elif get_platform()[-4:] == 'i386' or 'intel' in get_platform() or \
                'x86_64' in get_platform() or \
                'i386' in platform.platform():
                 intel = 1
-            else:
-                intel = 0
             if (os.path.exists('/System/Library/Frameworks'
                               '/Accelerate.framework/') and
                     'accelerate' in libraries):
-                if intel:
+                if arm:
+                    pass
+                elif intel:
                     args.extend(['-msse3'])
                 else:
                     args.extend(['-faltivec'])
                 args.extend([
-                    '-I/System/Library/Frameworks/vecLib.framework/Headers'])
+                    '-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Headers'])
```

Comment:
This looks like it's going to break things. For me, the path is slightly different: `Versions/A/Frameworks` is `Versions/Frameworks` (no `A/`) for me in macOS 10.14

Also the `MacOSX.sdk` path I'm not sure about. I see:
```
DriverKit19.0.sdk	MacOSX.sdk		MacOSX10.15.sdk
```
there. Does this work for different SDK versions?

Reply:
Is there an explanation somewhere about why this is the correct path?

Reply:
I've learned a lot about this code section, my project now on python27 so the newest v1.21.0 cannot suit my demand.

Here is my env.
Hardware:
1. Chip:	Apple M1

Software:
1. macOS 11.4(20F71)
2. Xcode 12.5 Build version 12E262 both IDE and command line.

`
ls -alh /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Accelerate.framework/Versions/                            
total 0
drwxr-xr-x  4 root  wheel   128B May 30 14:37 .
drwxr-xr-x  7 root  wheel   224B May 30 14:37 ..
drwxr-xr-x  6 root  wheel   192B Mar 16 22:04 A
lrwxr-xr-x  1 root  wheel     1B May 30 14:25 Current -> A

`

`
locate 'vecLib.framework/Headers'|grep -v .h
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/BNNS
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/LinearAlgebra
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/Quadrature
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/Sparse
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/BNNS
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/LinearAlgebra
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/Quadrature
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/Sparse
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Headers
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Headers
/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Headers
/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Headers

`

Maybe the chaos macOS SDK path leads to this.
I think both the old and the new path cannot solve this build warning on every macOS version.
The wrong header path will take no effect build but error.